modules/react/common/lib/utils/components.ts

```diff
@@ -78,26 +76,43 @@ type ExtractHTMLAttributes<
  * @example
  * interface MyProps extends ExtractProps<typeof Card.Body> {}
  *
- * ExtractProps<typeof Card>; // React.ClassAttributes<HTMLDivElement> & React.HTMLAttributes<HTMLDivElement> & CardProps
- * ExtractProps<typeof Card, 'aside'>; // React.ClassAttributes<HTMLElement> & React.HTMLAttributes<HTMLElement> & CardProps
```

Comment:
I ended up removing `React.ClassAttributes` from the type which removes `ref` and `key`. `ref` is added by `ElementComponent` and depends on the `as` prop, and `key` is already available via JSX typings and cannot be accessed internally anyways.